modules/activitypub/user_settings.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package activitypub
+
+import (
+	user_model "code.gitea.io/gitea/models/user"
+)
+
+// GetKeyPair function
+func GetKeyPair(user *user_model.User) (pub, priv string, err error) {
+	var settings map[string]*user_model.Setting
+	if settings, err = user_model.GetUserSettings(user.ID, []string{"activitypub_privpem", "activitypub_pubpem"}); err != nil {
+		return
+	} else if len(settings) == 0 {
+		if priv, pub, err = GenerateKeyPair(); err != nil {
+			return
+		}
+		if err = user_model.SetUserSetting(user.ID, "activitypub_privpem", priv); err != nil {
+			return
+		}
+		if err = user_model.SetUserSetting(user.ID, "activitypub_pubpem", pub); err != nil {
+			return
+		}
```

Comment:
These methods should use a context (with a transaction).

We should define a schema for the key names. I like such names separated with dots like in our translation files.

Reply:
> We should define a schema for the key names. I like such names separated with dots like in our translation files.

That's a problem now, we already are use user settings in the current codebase so defining a scheme seems not a viable option anymore. See https://github.com/go-gitea/gitea/pull/17414#discussion_r754804759 I'm not sure if we actually agreed on anything from that comment, but it feels like we currently stick to the existing key names that are being used 🤷🏽

Reply:
Are we going to just stick with using key names instead of a schema for now?